coralproject/talk
# Content Security Policy + Nunjacks

## What does this PR do?

- Adds support for the [Content Security Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) header which is disabled by default
- Changes out the view templates from [`ejs`](https://www.npmjs.com/package/ejs) to [`nunjacks`](https://www.npmjs.com/package/nunjucks)
- New configuration parameter: **TALK_ENABLE_STRICT_CSP**
  - Setting this to `TRUE` will enforce the [Content Security Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) (or CSP). By default, this configuration is set to [report only](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP#Testing_your_policy) where the policy is not enforced, but any violations are reported to a provided URI. (Default `FALSE`)

## How do I test this PR?

- Try loading Talk:
  - In development
  - On different url's
  - By inserting an image from an external site into the infobox, see it doesn't load!
